opensearch/src/main/java/org/opensearch/sql/opensearch/data/type/OpenSearchDateType.java

```diff
@@ -290,9 +398,9 @@ public boolean shouldCast(ExprType other) {
 
   @Override
   protected OpenSearchDataType cloneEmpty() {
-    if (this.formatString.isEmpty()) {
-      return OpenSearchDateType.of(this.exprCoreType);
+    if (formats.isEmpty()) {
+      return OpenSearchDateType.of(exprCoreType);
     }
-    return OpenSearchDateType.of(this.formatString);
+    return OpenSearchDateType.of(String.join(" || ", formats));
```

Comment:
Will spaces before and after `||` affect formatting? I remember in the past we had an issue with this.

Reply:
No - I trim them (line 173).